syntax/leaderf.vim

```diff
@@ -13,6 +13,37 @@ if exists("b:current_syntax")
     finish
 endif
 
+
+let s:icons = []
+
+
+function! s:get_all_icons() abort
+    if !exists('s:icons') || empty(s:icons)
+        let l:icons = []
+
+        let l:exact_icons = values(get(g:, 'WebDevIconsUnicodeDecorateFileNodesExactSymbols', {}))
+        let l:ext_icons = values(get(g:, 'WebDevIconsUnicodeDecorateFileNodesExtensionSymbols', {}))
+        let l:default_icon = get(g:, 'WebDevIconsUnicodeDecorateFileNodesDefaultSymbol', '')
+
+        call extend(l:icons, l:exact_icons + l:ext_icons)
+        call add(l:icons, l:default_icon)
+        let s:icons = uniq(l:icons)
+    endif
+    return s:icons
+endfunction
+
+
+function! s:add_syntax_hi_devicons() abort
+    let l:group = 'Lf_hl_icon'
+    for l:icon in s:get_all_icons()
+        execute printf('syntax match %s /%s/', l:group, l:icon)
+    endfor
+
+    let l:font = get(g:, 'Lf_DevIconsFont', 'NONE')
+    execute printf("hi %s font='%s' gui=NONE guifg=NONE guibg=NONE cterm=NONE ctermfg=NONE ctermbg=NONE", l:group, l:font)
```

Comment:
I'm considering whether to make the icon colorful and the color can be customized.

Reply:
It's good.
I think it would be even better visually if you could change the colors.

Reply:
OK, then add the color feature in the future.

Reply:
Is it better to have the `Lf_hl_devIcons` highlight instead of the `g:Lf_DevIconsFont` option?

Set it as follows.
```vim
hi Lf_hl_devIcons ctermfg=23 ctermbg=255 guifg=#494b53 guibg=#fafafa font=Cica:h10:cSHIFTJIS:qDRAFT
```

Reply:
I want that different icon has different color.

Reply:
Does that mean that you have to set a color for each icon?

Reply:
For some frequently-used file types.

Reply:
Should we be able to define and configure highlights like `Lf_hl_devIcons_vim` or `Lf_hl_devIcons_py`?

Reply:
The icon is customizable.
What do you think?
Refer 175cd30fadde8c541d9d36aac1b474d234d69ae0.